### ISSUE TITLE
Indent comment indentation: Comments can align with previous/next lines of code even if whitespace in between

### ISSUE THREAD

**Issue by platinumazure** [eslint/eslint#9733]
**Tell us about your environment**

* **ESLint Version:** 4.13.1 (via online demo)
* **Node Version:** N/A
* **npm Version:** N/A

**What parser (default, Babel-ESLint, etc.) are you using?**

Default parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
    "rules": {
        "indent": ["error", 4]
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
if (foo) {

// blah
    statement();

}
```

**What did you expect to happen?**

3:1 - Expected indentation of 4 spaces but found 0. (indent)

**What actually happened? Please include the actual, raw output from ESLint.**

No lint error.

------------

The bug (as I see it) is that the indent rule has a tolerance for comments to match the indentation of the previous line, that of the next line, or the "actual correct location". However, I think this should only apply if there actually is code on the *immediately previous or next lines*, rather than code that is separated by one or more lines of whitespace.

**Comment by platinumazure** [comment]
@not-an-aardvark: Is this a bug, or working as designed? Thanks!

**Comment by not-an-aardvark** [comment]
I think this is technically working as designed, but I agree with your analysis that the design could be improved here by making it more strict. I'm fine with calling it a bug.

**Comment by platinumazure** [comment]
I'll work on this later this week.

**Comment by sstern6** [comment]
@platinumazure was wondering if you were still looking into this one?  Wouldnt mind taking a crack at it.  If youre still looking into it though no worries.

**Comment by platinumazure** [comment]
Thanks for following up! I haven't had a chance to look into this; please
take a look if you are interested.

**Comment by platinumazure** [comment]
Working on this.
